pymongoexplain/commands.py

```diff
@@ -24,25 +24,32 @@
 Document = Union[dict, SON]
 
 class BaseCommand():
-    def __init__(self, dictionary):
-        if type(self) == BaseCommand:
-            raise NotImplementedError
-        self.command_document = self.convert_to_camelcase(dictionary)
-
-    def convert_to_camelcase(self, d: dict) -> dict:
+    def __init__(self, command_name, collection):
+        self.command_document = {}
```

Comment:
I think the `BaseCommand` class should define a standard API for adding new commands. In the current design, new commands need everything to be done in `__init__` which isn't particularly intuitive. Also, it doesn't strike me as particularly great design how we need to pass in the `command_name` to the init of the base class - this kind of a pattern should be handled via attributes on the class itself. How about something like this:

```
class BaseCommand:
    def __init__(self, collection, *args, **kwargs):
        self.collection = collection
        self.command_document = self._get_command_document(*args, **kwargs)

    @property
    def command_name(self):
        """New command classes will specify the command name here."""
        return None

    def _get_command_document(self, *args, **kwargs):
        """New command classes will override this method."""
        raise NotImplementedError   


class UpdateCommand(BaseCommand):
    @property
    def command_name(self):
        return "update"

    def _get_command_document(self, filter, update, kwargs):
        cmd_doc = {"updates":[{"q": filter, "u": update}]}
        for key, value in kwargs.items():
            ...
            ...
        return snake_to_camelcase(cmd_doc)
```

Reply:
Done.

Reply:
I agree that there are some improvements we can make here. Some minor exceptions I have
- I slightly prefer `BaseCommand. __init__` to receive the command `name` as an argument instead of through a property.
- I'm not a fan of how `*args, **kwargs` is used in your example. I think it's better for each class to define the arguments explicitly. 

What do you think about opening a new issue for improvement rather than refactoring this internal API in this review?

Reply:
Oops, I already did this change. Should I revert?

Reply:
We can make the improvements Shane suggested in a subsequent PR as recommended.